rotkehlchen/api/rest.py

```diff
@@ -2297,6 +2311,106 @@ def get_asset_icon(
 
         return response
 
+    @staticmethod
+    def _get_current_assets_price(
+            assets: List[Asset],
+            target_asset: Asset,
+    ) -> Dict[str, Any]:
+        """Return the current price of the assets in the target asset currency.
+        """
+        log.debug(
+            f'Querying the current {target_asset.identifier} price of these assets: '
+            f'{", ".join([asset.identifier for asset in assets])}',
+        )
+        assets_price = {asset: Inquirer().find_usd_price(asset) for asset in assets}
+        if target_asset != A_USD:
```

Comment:
Hmm that's really bad. You are making `N` more queries here where `N` is the number of assets in the list.

You should instead implement a generic current price query in inquirer. It's dead simple. The `find_usd_price` calls both cryptocompare and coingecko with target being `USD`. You just parametrize that.

We don't have it yet because it was not needed.